server/channels/api4/plugin.go

```diff
@@ -58,6 +58,10 @@ func uploadPlugin(c *Context, w http.ResponseWriter, r *http.Request) {
 	}
 
 	if err := r.ParseMultipartForm(MaximumPluginFileSize); err != nil {
+		if err.Error() == "http: request body too large" {
+			c.Err = model.NewAppError("uploadPlugin", "api.plugin.upload.file_too_large.app_error", nil, "", http.StatusRequestEntityTooLarge)
```

Comment:
Not a requirement, but It would be good to display the reached limit and how big the plugin's size is too

Unfortunate that this is hardcoded to 50MB. @hanzei Are you aware of this hardcoded limit? I thought this would follow the regular "max file upload size" server config variable

Reply:
The argument passed to `ParseMultipartForm` is the maximum number of bytes that are fetched into memory. (https://pkg.go.dev/net/http#Request.ParseMultipartForm) It does not limit the total size of the plugin.